mlflow/pyfunc/__init__.py

```diff
@@ -627,6 +628,12 @@ def load_model(
             f'Model does not have the "{FLAVOR_NAME}" flavor',
             RESOURCE_DOES_NOT_EXIST,
         )
+    if os.path.exists(os.path.join(local_path, FEATURE_SPEC_RELATIVE_FILE_PATH)):
```

Comment:
Could we instead check the model flavor, and call into a helper function shared by https://github.com/mlflow/mlflow/blob/42b2dcb45c52e6df58956851961f251493f245bf/mlflow/store/_unity_catalog/registry/rest_store.py#L142?